test/async/http/endpoint.rb

```diff
@@ -81,6 +81,18 @@
 				expect(subject).not.to be(:secure?)
 			end
 		end
+		
+		with 'invalid scheme' do
+			it "should raise an argument error" do
+				expect do
+					Async::HTTP::Endpoint.for("foo", "localhost")
+				end.to raise_exception(ArgumentError, message: be =~ /scheme/)
```

Comment:
I'm a fan of testing the full exception message to ensure it includes the correct text and values.